kube/src/client/verb.rs

```diff
@@ -0,0 +1,346 @@
+    params::Patch<Kind>: Serialize,
+{
+    type ResponseDecoder = DecodeSingle<Kind>;
+
+    fn to_http_request(&self) -> Result<Request<Body>> {
+        Request::patch(format!(
+            "{}/{}",
+            Kind::url_path(&self.dyn_type, self.scope.namespace()),
+            self.name
+        ))
+        .body(Body::from(
+            serde_json::to_vec(self.patch).context(SerializeFailed)?,
+        ))
+        .context(BuildRequestFailed)
+    }
+}
+
+/// Replace a named existing object
+pub struct Replace<'a, Kind: Resource, Scope> {
+    /// The object to replace
+    /// Requires `metadata.resourceVersion` to be `Some`
+    pub object: &'a Kind,
+    /// The scope of the object
+    pub scope: &'a Scope,
+    /// The type of the objects
+    pub dyn_type: &'a Kind::DynamicType,
+}
+impl<'a, Kind: Resource + Serialize + DeserializeOwned, Scope: scope::Scope> Verb
+    for Replace<'a, Kind, Scope>
+{
+    type ResponseDecoder = DecodeSingle<Kind>;
+
+    fn to_http_request(&self) -> Result<Request<Body>> {
+        Request::patch(format!(
+            "{}/{}",
+            Kind::url_path(&self.dyn_type, self.scope.namespace()),
+            self.object.meta().name.as_ref().context(UnnamedObject)?
+        ))
+        .body(Body::from(
+            serde_json::to_vec(self.object).context(SerializeFailed)?,
+        ))
+        .context(BuildRequestFailed)
+    }
+}
+
+/// Get the API server's version
+pub struct GetApiserverVersion;
+impl Verb for GetApiserverVersion {
+    type ResponseDecoder = DecodeSingle<k8s_openapi::apimachinery::pkg::version::Info>;
+
+    fn to_http_request(&self) -> Result<Request<Body>> {
+        Request::get("/version")
+            .body(Body::empty())
+            .context(BuildRequestFailed)
+    }
+}
+
+/// Get all API groups supported by the API server
+pub struct ListApiGroups;
+impl Verb for ListApiGroups {
+    type ResponseDecoder = DecodeSingle<k8s_openapi::apimachinery::pkg::apis::meta::v1::APIGroupList>;
+
+    fn to_http_request(&self) -> Result<Request<Body>> {
+        Request::get("/apis")
+            .body(Body::empty())
+            .context(BuildRequestFailed)
+    }
+}
+
+/// Get all supported versions of the legacy core API group
+pub struct ListCoreApiVersions;
+impl Verb for ListCoreApiVersions {
+    type ResponseDecoder = DecodeSingle<k8s_openapi::apimachinery::pkg::apis::meta::v1::APIVersions>;
+
+    fn to_http_request(&self) -> Result<Request<Body>> {
+        Request::get("/api")
+            .body(Body::empty())
+            .context(BuildRequestFailed)
+    }
+}
+
+/// Get all resources supported by the API server for a given API group and version
+pub struct ListApiGroupResources<'a> {
+    /// The API group, use `""` for the legacy core group
+    pub group: &'a str,
+    /// THe API version
+    pub version: &'a str,
+}
+impl<'a> Verb for ListApiGroupResources<'a> {
+    type ResponseDecoder = DecodeSingle<k8s_openapi::apimachinery::pkg::apis::meta::v1::APIResourceList>;
```

Comment:
now you are getting into discovery - which is arguably a list on the APIResource resource, rather than a separate verb (but legacy resources are awkward).

Reply:
True, I wonder if we could add a special-case impl on `List<APIResource>` without conflicting with the blanket...

Reply:
Hm, probably not actually, since `List` doesn't have the information about the group and version.